Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+			m_Frame = (m_Frame + 1) % m_FrameCount;
+			m_SpriteAnimTimer.Reset();
+		}
+
+		// Lose health when door is lost, spinning out of control until grinds to halt
+		if (!m_Door && m_Status != DYING && m_Status != DEAD) {
+			m_SpriteAnimMode = ALWAYSLOOP;
+			m_SpriteAnimDuration = LERP(0, m_MaxHealth, 10, m_InitialSpriteAnimDuration, m_Health);
+
+			// TODO: Figure out how to pitch this correctly so it goes up as SpriteAnimDuration goes down.
+			/*
+			if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+			g_AudioMan.SetSoundPitch(&m_DoorMoveSound, LERP(0, m_InitialSpriteAnimDuration, 0, 5, m_SpriteAnimDuration));
+			*/
+
+			m_Health -= 0.4F;
+		}
+		if (m_Status == DEAD) { GibThis(Vector(), 50); }
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::UpdateSensors() {		
+		const Actor *foundActor = 0;
+		bool anySensorInput = false;
+
+		for (ADSensor &sensor : m_Sensors) {
+			foundActor = sensor.SenseActor(m_Pos, m_Rotation, m_HFlipped, m_MOID);
+			if (foundActor && foundActor->IsControllable()) {
+				anySensorInput = true;
+				// Look for enemy actors first and break the loop so we don't accidentally open the door for a friendly actor is enemies are nearby.
+				if (foundActor->GetTeam() != m_Team) {
+					CloseDoor();
+					break;
+				} else if (foundActor->GetTeam() == m_Team) {					
+					OpenDoor();
+				}
+			}
+		}
+		// If no sensor input, return to default state if it's time. This won't override any actions that the sensors started
+		if (!anySensorInput && m_ResetDefaultTimer.IsPastSimMS(m_ResetDefaultDelay)) {
+			if (m_ClosedByDefault && m_DoorState == OPEN) {
+				CloseDoor();
+			} else if (!m_ClosedByDefault && m_DoorState == CLOSED) {
+				OpenDoor();
+			}
+		}
+		m_SensorTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::UpdateDoorAttachable() {
+		m_Door->SetHFlipped(m_HFlipped);
+		switch (m_DoorState) {
+			case OPEN:
+				m_Door->SetJointPos(m_Pos + m_OpenOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+				m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_OpenAngle * GetFlipFactor()));
+				break;
+			case OPENING:
+				if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+
+				if (m_DoorMoveTimer.IsPastSimMS(m_DoorMoveTime)) {
+					m_ResetDefaultTimer.Reset();
+
+					m_Door->SetJointPos(m_Pos + m_OpenOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_OpenAngle * GetFlipFactor()));
+
+					// Draw the material representation of the door to the terrain in the open state
+					if (m_DrawMaterialLayerWhenOpen) { DrawDoorMaterial(); }
+
+					m_DoorMoveSound.Stop();
+					m_DoorMoveEndSound.Play(m_Pos);
+
+					m_DoorState = OPEN;
+				} else {
+					// Find interpolated positions and angles for opening
```

Comment:
This comment is pretty pointless. I wouldn't be against one that actually explains things, but this is just data's garbage fucking habit (that I'd hope he's since kicked) of writing comments that are just the code but in English.